pkg/remote/aws/db_instance_supplier.go

```diff
@@ -43,8 +51,11 @@ func (s DBInstanceSupplier) Resources() ([]resource.Resource, error) {
 	resourceList, err := listAwsDBInstances(s.client)
 
 	if err != nil {
-		logrus.Error(err)
-		return nil, err
+		handled := handleListAwsError(err, resourceaws.AwsDbInstanceResourceType, s.alerter)
+		if !handled {
+			return nil, err
+		}
+		return []resource.Resource{}, nil
```

Comment:
Can we return err in any case there and handle error in the scanner more globally ? 
We can use a custom error to add local scope context like the resource type.

I know it changes original architecture of this PR but there will be a single point for error handling.

We could do something like this in supplier 

```go
if err != nil {
  return errorWithContext(err, resourceaws.AwsDbInstanceResourceType)
}
```

and in the parent call on scanner side

```go
	res, err := supplier.Resources()
	if err != nil {
		if handleListAwsError(err, s.alerter) {
			return nil, nil // or maybe return []resource.Resource{}, nil
		}
		return nil, err
	}
```

Reply:
hmm That sounds nice I'll look into this !

Reply:
last push is something with that idea please tell me what you think